RNA-seq/06-openpbta_heatmap.Rmd

```diff
@@ -0,0 +1,430 @@
+
+```{r create_annotation, live = TRUE}
+column_annotation <- HeatmapAnnotation(
+  df = sample_annotation_df,
+  col = sample_annotation_colors
+)
+```
+
+### Values for display
+
+We will z-score the expression values for display.
+This is sometimes called a standardized score.
+It's calculated for each value in a row (gene) by subtracting the gene's mean and dividing by the gene's standard deviation.
+
+```{r}
+zscores_mat <- 
+  (high_var_mat - rowMeans(high_var_mat)) / matrixStats::rowSds(high_var_mat)
+```
+
+
+Let's look at the mean and standard deviation of values for a randomly selected gene prior to standardization.
+
+```{r random_row_mean}
+mean(high_var_mat[25, ])
+```
+
+```{r random_row_sd}
+sd(high_var_mat[25, ])
+```
+
+And now those values after standardization.
+
+```{r random_z_mean, live = TRUE}
+mean(zscores_mat[25, ])
+```
+
+This value isn't exactly zero but it's very close!
+
+```{r random_z_sd, live = TRUE}
+sd(zscores_mat[25, ])
+```
+
+The standard deviation is now 1.
+That means all the variance values will be equal to one, too.
+
+```{r summary_row_variance}
+summary(matrixStats::rowVars(zscores_mat))
+```
+
+If you're filtering to high variance values, it's important to do that _prior_ to standardizing.
+
+### Heatmap itself!
+
+Okay, now we're ready to make a heatmap complete with annotation bars.
+
+```{r make_heatmap}
+Heatmap(zscores_mat,
+        # The distance metric used for clustering the rows
+        # This is different from the default (Euclidean)
+        clustering_distance_rows = "pearson",
+        # Linkage method for row clustering
+        # This is different from the default (complete)
+        clustering_method_rows = "average",
+        # Distance metric for columns
+        clustering_distance_columns = "pearson",
+        # Linkage for columns
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        # Add annotation bars to the top of the 
+        top_annotation = column_annotation,
+        # This will be used as the label for the color bar
+        # of the cells of the heatmap itself
+        name = "z-score")
+```
+
+We have to create the heatmap _twice_ -- once for display in the notebook and once to save to a PNG file.
+
+```{r save_heatmap_png, live = TRUE}
+# Open PNG plot device
+png(filename = heatmap_file, 
+    width = 11, 
+    height = 7, 
+    units = "in", 
+    res = 300)
+# Heatmap, again!
+Heatmap(zscores_mat,
+        clustering_distance_rows = "pearson",
+        clustering_method_rows = "average",
+        clustering_distance_columns = "pearson",
+        clustering_method_columns = "average",
+        show_row_names = FALSE,
+        show_column_names = FALSE,
+        top_annotation = column_annotation,
+        name = "z-score",
+        use_raster = FALSE) # higher resolution for output (be careful with PDF output!)
+# Shut down current graphics device
+dev.off()
+```
+
+### Notes on clustering
```

Comment:
Yep, I think this might be too long as is. BUT! We should also move your nice explanation of VST out of this notebook in my opinion. If we do that, do you still want do the transformation during the instruction or can we cooking show magic it 🍳 ?